File: daemons/macos/src/Handlers/DebugHandler.swift
import Foundation

enum DebugHandler {
    static func uploadIOSLog(_ request: HTTPRequest) -> HTTPResponse {
        let url = destinationURL
        try? FileManager.default.createDirectory(
            at: url.deletingLastPathComponent(), withIntermediateDirectories: true)
        if let object = try? JSONSerialization.jsonObject(with: request.body) as? [String: Any],
            let content = object["content"] as? String
        {
            try? Data(content.utf8).write(to: url, options: .atomic)
            return HTTPResponse.json(200, ["ok": true, "path": url.path])
        }
        return HTTPResponse.json(400, ["error": "bad_request"])
    }

    static var destinationURL: URL {
        FileManager.default
            .urls(for: .applicationSupportDirectory, in: .userDomainMask)[0]
            .appendingPathComponent("Cloude")
            .appendingPathComponent("ios-logs")
            .appendingPathComponent("latest.log")
    }
}

File: daemons/linux/src/Handlers/DebugHandler.js
import fs from 'node:fs'
import os from 'node:os'
import path from 'node:path'
import HTTPResponse from '../Networking/HTTPResponse.js'

const dataDirectory = process.env.CLOUDE_DATA || path.join(os.homedir(), '.cloude-agent')
const destinationPath = path.join(dataDirectory, 'ios-logs', 'latest.log')

function parsedBody(request) {
  try {
    return JSON.parse(request.body.toString('utf8'))
  } catch {
    return null
  }
}

export function uploadIOSLog(request) {
  const object = parsedBody(request)
  if (object && typeof object.content === 'string') {
    fs.mkdirSync(path.dirname(destinationPath), { recursive: true })
    fs.writeFileSync(destinationPath, object.content)
    return HTTPResponse.json(200, { ok: true, path: destinationPath })
  }
  return HTTPResponse.json(400, { error: 'bad_request' })
}
